.github/workflows/check-submodule.yml

```diff
@@ -0,0 +1,29 @@
+name: Check meta-mistysom Submodule
+
+on:
+  push:
+    branches: [ "master", "develop" ]
+  pull_request:
+    branches: [ "master", "develop" ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  check:
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v3
+
+      - name: Check SHA hash of submodule and the master/develop branch
+        run: |
+          cd $GITHUB_WORKSPACE/Build/meta-mistysom
+          SOM_HEAD = $(git rev-parse HEAD)
+          echo "The submodule is set to $SOM_HEAD\n"
+          SOM_ORIGIN = $(git rev-parse origin/$GITHUB_REF_NAME)
+          echo "The origin/$GITHUB_REF_NAME is set to $SOM_ORIGIN"
+          [ $SOM_HEAD == SOM_ORIGIN ]
+        shell: bash
```

Comment:
will this fail if the branches misalign?

Reply:
On paper yes. But it is failing before it reaches it. Still testing.

Reply:
Sounds good! Keep testing and let me know when it works! :)